docs/concepts/hashing.md

```diff
@@ -49,6 +49,60 @@ Cryptographic hashes come with a couple of very important characteristics:
+
+Let us look at a concrete example. When you download an image file for Ubuntu Linux you might see the following `SHA-256` checksum on the Ubuntu website listed for verification purposes:
+
+```
+0xB45165ED3CD437B9FFAD02A2AAD22A4DDC69162470E2622982889CE5826F6E3D ubuntu-20.04.1-desktop-amd64.iso
+```
+
+After downloading the Ubuntu image, you can verify the integrity of the file by hashing the file to make sure the checksums match:
+
+```shell
+echo "b45165ed3cd437b9ffad02a2aad22a4ddc69162470e2622982889ce5826f6e3d *ubuntu-20.04.1-desktop-amd64.iso" | shasum -a 256 --check
+
+ubuntu-20.04.1-desktop-amd64.iso: OK
+```
+
+If we add the `ubuntu-20.04.1-desktop-amd64.iso` file to IPFS we receive a hash as an output:
+
+```shell
+ipfs add ubuntu-20.04.1-desktop-amd64.iso
+
+added QmPK1s3pNYLi9ERiq3BDxKa4XosgWwFRQUydHUtz4YgpqB ubuntu-20.04.1-desktop-amd64.iso
+ 2.59 GiB / 2.59 GiB [==========================================================================================] 100.00%
+```
+
+The string `QmPK1s3pNYLi9ERiq3BDxKa4XosgWwFRQUydHUtz4YgpqB` returned by the `ipfs add` command is the content identifier of the file `ubuntu-20.04.1-desktop-amd64.iso`. We can utilize the [CID Inspector](https://cid.ipfs.io/) to see what the CID includes. The actual hash is listed under `DIGEST (HEX)`:
+
+```
+NAME: sha2-256
+BITS: 256
+DIGEST (HEX): 0E7071C59DF3B9454D1D18A15270AA36D54F89606A576DC621757AFD44AD1D2E
+```
+
+::: tip
+The names of hash functions are not used consistently.`SHA-2`, `SHA-256` or `SHA-256 bit` all refer to the same hash function.
+:::
+
+We can now check if the hash contained in the CID equals the checksum for the file:
+
+```shell
+echo "0E7071C59DF3B9454D1D18A15270AA36D54F89606A576DC621757AFD44AD1D2E *ubuntu-20.04.1-desktop-amd64.iso" | shasum -a 256 --check
+
+ubuntu-20.04.1-desktop-amd64.iso: FAILED
+shasum: WARNING: 1 computed checksum did NOT match
+```
+
+As we can see the hash included in the CID does NOT match the hash of the input file `ubuntu-20.04.1-desktop-amd64.iso`. To understand what the hash included in the CID actually is, we have to understand how IPFS stores files. IPFS uses a so called Distributed Acyclic Graph (DAG) to keep track of all the data stored in IPFS. A CID identifies one specific node in this graph. This identifier is the result of hashing the node's contents — any opaque payload carried by the node and the list of identifiers of its children — using a cryptographic hash function like `SHA256`. To learn more about wow the DAG is used within IPFS to address content is explained in detail in the article covering the [Merkle DAG](/concepts/merkle-dag/).
+
+::: tip
+The [Tutorial: Merkle Trees and the IPFS DAG](https://dweb-primer.ipfs.io/ipfs-dag) inlcudes in-depth lessons coveing the IPFS DAG.
+:::
```

Comment:
```suggestion
```

I'm reluctant to include any links to IPFS Primer docs. They're pretty outdated. This particular link was last updated 12 months ago.